src/modules/storage/LedgerStorage.ts

```diff
@@ -297,13 +308,65 @@ export class LedgerStorage extends Storages
             });
         }
 
+        /**
+         * Saving Average Fees
+         */
+        async function save_fees(storage: LedgerStorage, block: Block, genesis_timestamp: number){            
+            async function calculateFee (txs: Transaction[])
+            {
+                    let total_tx_fee = 0, total_payload_fee=0, total_fee=0, tx_count=0;
+                    for (let index = 0; index < block.txs.length; index++) {
+                        if (!(block.txs[index].isCoinbase())) {
+                            const fees = await storage.getTransactionFee(block.txs[index]);
+                                if (JSBI.toNumber(fees[1])>=0) {
+                                    total_tx_fee += JSBI.toNumber(fees[1]);
+                                    total_payload_fee += JSBI.toNumber(fees[2]);
+                                    total_fee += JSBI.toNumber(fees[0]);
+                                    tx_count++;
+                                }
+                        }
+                    }
+                return {total_tx_fee, total_payload_fee, total_fee, tx_count}
+            }
+            const fee = await calculateFee(block.txs)
+            const tx_count = (fee.tx_count!=0)?fee.tx_count:1
+            const average_tx_fee = fee.total_tx_fee/tx_count
```

Comment:
Can we calculate in transaction bytes to show useful information?
Average transaction fee = bytes per BOA

Reply:
SUM(a Tx fee / a Tx size) / All Txs count  = BOA per bytes